templates/test/ci/patches/windows-server-version.yaml

```diff
@@ -6,4 +6,8 @@ spec:
   template:
     metadata:
       annotations:
-        windowsServerVersion: ${WINDOWS_SERVER_VERSION:=""}
\ No newline at end of file
+        windowsServerVersion: ${WINDOWS_SERVER_VERSION:=""}
+    spec:
+      identity: UserAssigned
+      userAssignedIdentities:
+      - providerID: /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${CI_RG:=capz-ci}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${USER_IDENTITY:=cloud-provider-user-identity}
```

Comment:
Leaving this as a reminder to follow-up after this PR and check to see if these new user assigned identities are necessary.

Reply:
As I found out last week, these mostly are necessary. There may be some corners we can cut for tests that don't involve cloud provider authenticating with Azure, but I'm not sure it's worth finding out exactly which ones those might be.